lms/djangoapps/verify_student/utils.py

```diff
@@ -93,28 +93,56 @@ def send_verification_status_email(context):
         ))
 
 
-def most_recent_verification(photo_id_verifications, sso_id_verifications, most_recent_key):
+def most_recent_verification(photo_id_verifications, sso_id_verifications, manual_id_verifications, most_recent_key):
```

Comment:
i wonder if there is a cleaner way to do this by accepting a list or dict of querysets - so we don't have to change this method if we ever add another type of verification object. as it is, the logic got considerably more complex, and it seems like this is an opportunity to simplify it to handle an arbitrary number of types of objects.

Reply:
@brittneyexline Kindly review https://github.com/edx/edx-platform/pull/18337#discussion_r194333246

I had previously coded this as:
```python
def most_recent_verification(photo_id_verifications, sso_id_verifications, manual_id_verifications, most_recent_key):
    """
    Return the most recent verification given querysets for photo, sso and manual verifications.

    Arguments:
            photo_id_verifications: Queryset containing photo verifications
            sso_id_verifications: Queryset containing sso verifications
            manual_id_verifications: Queryset containing manual verifications
            most_recent_key: Either 'updated_at' or 'created_at'

    Returns:
        The most recent verification.
    """
    photo_id_verification = photo_id_verifications and photo_id_verifications.first()
    sso_id_verification = sso_id_verifications and sso_id_verifications.first()
    manual_id_verification = manual_id_verifications and manual_id_verifications.first()

    verifications = [photo_id_verification, sso_id_verification, manual_id_verification]

    verifications_map = {
        verification: getattr(verification, most_recent_key)
        for verification in verifications
        if getattr(verification, most_recent_key, False)
    }

    return max(verifications_map, key=lambda k: verifications_map[k]) if verifications_map else None

```

Reply:
I see. I think this is more concise and extensible if more models come along... perhaps explaining the logic with a comment since it isn't so obvious when you read it would suffice. This could even be further simplified if the argument is changed to a list or dict so that we don't have to change the method signature again either.

This is just my opinion... @zubair-arbi and @adamstankiewicz do you think adding a comment to explain the logic will address your readability concerns? I don't find the if/elif block more readable, personally.

Reply:
@brittneyexline @zubair-arbi @irfanuddinahmad After seeing the `if/elif` block with the `manual_id_verification`, I think it does make sense to switch over to how @irfanuddinahmad proposes it to support more models.

I think adding a comment to help explain the logic would be sufficient.

Reply:
@irfanuddinahmad please use your previous compact code with a comment to briefly explain the logic.